### PR TITLE
Fix python package issue and upgrade the linux image to 2010

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

1. Increase job timeout, while we are investigating why the tests take much longer
2. Upgrade the linux docker image to manylinux2010, by request from Tianlei. (We had an offline discussion with Pranav and Tracy)
3. Remove the installation of "devtoolset-7" in the CUDA image. It was added for CUDA 10.0, it is not needed for CUDA 10.1. We have moved to CUDA 10.1.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu [hunk 1]
```diff
@@ -7,7 +7,6 @@ ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
 
 ADD scripts /tmp/scripts
 RUN cd /tmp/scripts && /tmp/scripts/install_centos.sh && /tmp/scripts/install_deps.sh -p $PYTHON_VERSION && \
-yum remove -y devtoolset-\*  && yum install -y devtoolset-7-binutils devtoolset-7-gcc devtoolset-7-gcc-c++ devtoolset-7-gcc-gfortran devtoolset-7-libquadmath-devel devtoolset-7-libstdc++-devel devtoolset-7-runtime && \
```
Comment (edgchen1) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu:10]: should tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu be updated too?

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu:10]: Where it is used?

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu:10]: manylinux2014 is a different story. I guess its gcc version is higher than the one(8.3) in manylinux 2010, and CUDA 10.1 may not work with it.

Comment (edgchen1) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu:10]: it's used by the training python package build

**Pull request merged**